Clear recent meeting list & Adjust number of recent meeting to display

Capataloutchoupika opened the pull request:
- Clear recent meeting list
- Adjustable number of recent meeting to display with interface_config.js

jitsi-jenkins commented:
Hi, thanks for your contribution!
If you haven't already done so, could you please make sure you sign our CLA (https://jitsi.org/icla for individuals and https://jitsi.org/ccla for corporations)? We would unfortunately be unable to merge your patch unless we have that piece :(.

hristoterezov commented on .gitignore:
```diff
@@ -6,6 +6,7 @@ all.css
 *css.map
 .remote-sync.json
 .sync-config.cson
+update_fork.sh
```
Could you please remove this?

hristoterezov commented on lang/main-fr.json:
```diff
@@ -746,6 +746,7 @@
         "privacy": "Confidentialité",
         "recentList": "Récent",
         "recentListDelete": "Supprimer",
+        "recentListDeleteAll": "Supprimer l'historique",
```
@damencho what is the status of the translations? Do we update the other languages manually?

Capataloutchoupika replied:
I don't know how you do usually, I can add the line in each file manually...

hristoterezov replied:
@damencho Will know I hope :) Please hold on this one!

damencho replied:
Just edit main.json if needed and do not touch the rest of the files, please. Thanks.

Capataloutchoupika replied:
Done ! :) I'll remember :)

hristoterezov commented on react/features/base/react/components/web/MeetingsList.js:
```diff
@@ -88,20 +95,38 @@ export default class MeetingsList extends Component<Props> {
+     */
+    _onClear() {
+        window.localStorage.setItem('features/recent-list', []);
+        window.location.reload();
```
I think we should handle this a little bit more gracefully. We already have an action that deletes a single entry from the list. Maybe you can create one for deleting all entries? I believe this will be auto synced with local storage and in the same time you won't need a reload to display the change.

Capataloutchoupika replied:
Yep, didn't see this action, it's only available on mobile app. I implemented a similar action to delete all the list (list entry delete).

hristoterezov commented on react/features/base/react/components/web/MeetingsList.js:
```diff
@@ -199,3 +237,14 @@ export default class MeetingsList extends Component<Props> {
+    return { };
+}
+
+export default translate(connect(_mapStateToProps)(MeetingsList));
```
Do you need connect at all? Also maybe we should keep the pattern here and pass the function that deletes everything as a prop. This way meetings list will be generic.

Capataloutchoupika replied:
I changed connect function without parameter, I don't know how to call prop function (delete or dispatch) without connect. Moreover dispatch actions are useful^^

hristoterezov replied:
See how we do it for onPress prop. That's what I mean. This way you won't need dispatch here. And this component will still be generic. And also it will be more consistent since we do this for onPress already.

hristoterezov replied:
I don't think this one is addressed. Do you plan to address it?

Capataloutchoupika replied:
In fact, I have no idea to do without that... i was inspired by the function deleteRecentListEntry() which is available on mobile to delete one entry. This function use dispatch and connect (to translate and dispatch). I also add this function the ReducerRegistry like other.
I you want a better solution, I can close this PR and think about a better solution with more knowledge about react.

hristoterezov replied:
No no don't close it! It just needs a very very small change.

Here's more detailed explanation about the change that I propose:
 - move `_onClear` method in react/features/recent-list/components/RecentList.web.js
 - pass the `_onClear` method to the `MeetingsList` component in the render method as a `onClear` property.
 - in react/features/base/react/components/web/MeetingsList.js use the `onClear` property value to execute the function when the new button is pressed. it is good to add a check if the property is not passed to hide the button
As a result:
 - You won't need connect in react/features/base/react/components/web/MeetingsList.js at all.
Purpose of the change:
MeetingList is used to just render whatever we pass to the component. All the implementation details are provided by the components that are using it. In this case `RecentList`.

Let me know if you want to try to do the change!

If not, would you allow me to push a commit to your PR? Note that it may take a while to clear my backlog of more urgent tasks until I get to this one!

Capataloutchoupika replied:
i'm trying this ! (sorry didn't have many time today... double IT crisis due to quarantine and VPN network overload :(). I'll update asap !

Capataloutchoupika replied:
Done ! :) thx a lot for these advices and explanations !

hristoterezov commented:
@saghul Do you think we should add this to mobile? I'm asking because of the interfaceConfig option for meetings length. If we are going to implement it on mobile maybe here we can use config prop instead of interfaceConfig prop which can be reused on mobile. Although I'm not sure if that will be working on mobile since I'm not sure if config is loaded on the landing page?

hristoterezov commented on react/features/base/react/components/web/MeetingsList.js:
```diff
@@ -88,20 +102,35 @@ export default class MeetingsList extends Component<Props> {
+    _onClear: void => void;
+
+    /**
+     * Clear all history items and refresh page.
```
It doesn't refresh the page anymore.

Capataloutchoupika replied:
removed from MeetingList.js

hristoterezov commented on react/features/recent-list/actions.js:
```diff
@@ -3,9 +3,25 @@
+/**
+ * Deletes a recent list entry based on url and date.
+ *
+ * @param {Object} entryId - An object constructed of the url and the date of
```
It doesn't have parameters and deletes the whole list, not an entry.

Capataloutchoupika replied:
Mybad ! wrong copy/paste !

Capataloutchoupika commented:
Update...

@hristoterezov thanks for your help and advices ! I updated all my files thanks to your tips (really easy) :
- move onClear function to RecentList.web.js and pass it as argument in MeetingsList prop
- Removed connect() on MeetingsList as wanted
- Comment correction for deleteRecentList() in actions.js
- Removed .meetings-list-web-clear > .hide class in css (now controlled by onClear function && meetings array size

Should be really better now !

Capataloutchoupika commented:
Hello @hristoterezov , do my latest changes suit you ? Or do I need to make some improvements ?

hristoterezov commented:
@Capataloutchoupika Thanks for the changes! No further improvement needed at this point! Will do some final tests later today and if everything works as intended will merge the PR.

luixxiul commented:
How about the status on this PR?

krombel commented:
In electron there was a PR to allow deletion of individual rooms (which is useful to only delete rooms you are sure you won't need anymore while keeping the "regular ones")

The implementation of that is linked directly above this comment.

What do you think of that approach?
